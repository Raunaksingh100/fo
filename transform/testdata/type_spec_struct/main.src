package main

type A[T] struct {
	a T
	b []T
	c [3]T
	d map[T]T
	e map[T]string
	f map[string]T
	g *T
	h ***T
	i chan T
	j []map[T]chan T
	k func(T)
}

type B[T, U] struct {
	a map[T]U
	b func(T) U
}

type C[T, U] struct {
	nested struct {
		a T
		b U
	}
}

func main() {

}
