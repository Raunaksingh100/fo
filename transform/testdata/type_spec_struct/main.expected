package main

type A struct {
	a interface{}
	b interface{}
	c interface{}
	d interface{}
	e interface{}
	f interface{}
	g interface{}
	h interface{}
	i interface{}
	j interface{}
	k interface{}
}

type B struct {
	a interface{}
	b interface{}
}

type C struct {
	nested struct {
		a interface{}
		b interface{}
	}
}

func main() {

}
